### DISCUSSION TITLE
Jan 7 Dev Session

### DISCUSSION THREAD

**Discussion by tpike3** [mesa/mesa#1553] [General]
When: 7 January --- 2000 UTC+8/1300 CET/1200 GMT/0700 EDT

Where: [via Matrix](https://matrix.to/#/#mesa-dev:matrix.org)

Proposed Agenda:

- @Logende cache results contributions
- Mesa 2.0
- Mesa- Geo 
- Tutorial Update

@jackiekazil @rht @Corvince @wang-boyu @Tortar @aiwhoo @nunezco2 @EwoutH  @logende Please feel free to add or takeaway from this agenda

**Comment by tpike3** [comment]
Reminder Mesa Dev Session this Saturday 7 January

**Comment by Logende** [comment]
I was really looking forward to this dev session but unfortunately I won’t be able to take part. Since I’m back from vacation, i still have quite high fever today and my ability of formulating consistent and useful thoughts and talking about them is strongly reduced.

Anyways, one part I would be interested in as one possible outputs of this session is, how you intend to work on Mesa-Replay until you decide its API now is stable enough and it has the desired improvements (such as more optimized cache size). Do you intend to create a fork of the repo, work on it and once it is far enough, manually copy the code into the main mesa repo, or do you have another idea?

**Reply by tpike3** [reply]
@Logende  That stinks, I hope your feeling better and I will try and make sure I get an answer to your question in the notes.

**Comment by jackiekazil** [comment]
We had a lot to talk about... I forgot to add discussion about permission patterns (from last month). We should add this to next month.

**Comment by tpike3** [comment]
## 7 January Notes
----------------------

#### Integrating New Functionality, specifically *Mesa - Replay* by @Logende

Background: When presenting results, computation can hinder ABMs, caching results so one can replay the model can be a highly effective approach to overcome this. @Logende built a great library to start addressing this challenge. The question for Project Mesa is how to keep momentum on such efforts and try and continue to improve them as it is a non-trivial challenge. 

The Dev Team agreed this functionality should be part of Core Mesa (i.e. main mesa repo) but needs to be developed more, as this is a very non-trivial problem. 

Way Ahead:  
- @jackiekazil  will fork the @Logende  repo into ProjectMesa
- ProjectMesa will work to advertise the package to get more use and support to aid is development this includes: 
     - @tpike3 Including in Monthly NUMFOCUS write- up 
     - @jackiekazil posting about on Mesa linkedin
     - @tpike3 highlighting it to Mesa email network (COMPLETE)
     - @jackiekazil added to [Mesa Packages](https://github.com/projectmesa/mesa/wiki/Mesa-Packages)
-  @Logende would you be interested in [writing a JOSS paper ](https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements)? (Rethink based on JOSS specifications) 

### Mesa Development

Dev team decided improvements to space.py (grid) is major change for Mesa 2.0; However, many changes can be made for a Mesa 1.2 release. The goal is Mesa 1.2 released at next Dev Session 4 Februrary

- Milestones have been updated to reflect goal milestones for Mesa 1.2 and Mesa 2.0 (please let me know if I need to reassign any or forgot any) 

### Additional Issues

- How to integrate examples into *Docs*? 
     - With new example repo what is the best way to to let user know about examples and make them accessible through documentation
 - Docs need to be improved so API is go to resource for users and not the code (added as issues for Mesa 2.0 #1580)
 - We still need to action permissions to improve Mesa resilience and security will be a topic for 4 Feb Dev session 

As always please feel free to correct or comment so we can make sure the notes are accurate.

Next Dev Session 4 February

**Reply by Logende** [reply]
That sounds very good! :) I appreciate the time you have put into addressing the future of the replay functionality and I feel like it is in good hands.
I am indeed interested in writing a JOSS Paper and will start doing so within the next days. 

Hopefully everything was clear regarding Mesa-replay during the session. If at any point you should have any question about it, feel free to reach out to me.

**Reply by Logende** [reply]
@tpike3 do Mesa and extensions such as Mesa-Geo have a JOSS Paper and if yes, where can I find them? I could not find any paper.md or paper.rst files in the repos.

Also, I've read the JOSS requirements, and mesa-replay is rather in the gray area. Some of the requirements are:

> As a rule of thumb, JOSS’ minimum allowable contribution should represent not less than three months of work for an individual. 


> Submissions under 1000 LOC will usually be flagged, those under 300 LOC will be desk rejected.


> Whether the software has already been cited in academic papers.
> Whether the software is sufficiently useful that it is likely to be cited by your peer group.


> In addition, JOSS requires that software should be feature-complete (i.e., no half-baked solutions), packaged appropriately according to common community standards for the programming language being used (e.g., [Python](https://packaging.python.org/), [R](https://r-pkgs.org/index.html)), and designed for maintainable extension (not one-off modifications of existing tools). “Minor utility” packages, including “thin” API clients, and single-function packages are not acceptable.

**Reply by tpike3** [reply]
@tpike3  Apologies, we have not attempted to publish in JOSS before and it looks like we need to let mesa-replay bake some more. I think I somehow got the wrong impression like some medical journals you could do a short write-up of emerging concepts. 

As the goal is to get the word out I will see if there is a more appropriate venue and make sure it is included in the news letters. Again apologies for the confusion.

**Reply by Logende** [reply]
> @tpike3 Apologies, we have not attempted to publish in JOSS before and it looks like we need to let mesa-replay bake some more. I think I somehow got the wrong impression like some medical journals you could do a short write-up of emerging concepts.
> 
> As the goal is to get the word out I will see if there is a more appropriate venue and make sure it is included in the news letters. Again apologies for the confusion.

Okay, yeah that makes sense. 

I would assume that Mesa itself would be ready for JOSS though, if someone would have the time to write the corresponding paper.

**Comment by jackiekazil** [comment]
We have two Mesa papers. Currently writing an update paper. My concern
about JOSS would be that it would cause confusion… but that also might put
a line in the sand as the ultimate paper.

Let me consider it for a week or two.
